sys/xtimer/xtimer_core.c

```diff
@@ -173,88 +145,13 @@ static inline void _lltimer_set(uint32_t target)
-    uint32_t now = _xtimer_now();
-    int res = 0;
-
-    timer->next = NULL;
-
-    /* Ensure that offset is bigger than 'XTIMER_BACKOFF',
-     * 'target - now' will always be the offset no matter if target < or > now.
-     *
-     * This expects that target was not set too close to now and overrun now, so
-     * from setting target up until the call of '_xtimer_now()' above now has not
-     * become equal or bigger than target.
-     * This is crucial when using low CPU frequencies so reaching the '_xtimer_now()'
-     * call needs multiple xtimer ticks.
-     *
-     * '_xtimer_set()' and `_xtimer_periodic_wakeup()` ensure this by already
-     * backing off for small values. */
-    uint32_t offset = (target - now);
-
-    DEBUG("timer_set_absolute(): now=%" PRIu32 " target=%" PRIu32 " offset=%" PRIu32 "\n",
-          now, target, offset);
-
-    if (offset <= XTIMER_BACKOFF) {
-        /* backoff */
-        xtimer_spin_until(target);
-        _shoot(timer);
-        return 0;
-    }
-
-    unsigned state = irq_disable();
-    if (_is_set(timer)) {
-        _remove(timer);
-    }
-
-    timer->target = target;
-    timer->long_target = _long_cnt;
-
-    /* Ensure timer is fired in right timer period.
-     * Backoff condition above ensures that 'target - XTIMER_OVERHEAD` is later
-     * than 'now', also for values when now will overflow and the value of target
-     * is smaller then now.
-     * If `target < XTIMER_OVERHEAD` the new target will be at the end of this
-     * 32bit period, as `target - XTIMER_OVERHEAD` is a big number instead of a
-     * small at the beginning of the next period. */
-    target = target - XTIMER_OVERHEAD;
-
-    /* 32 bit target overflow, target is in next 32bit period */
-    if (target < now) {
-        timer->long_target++;
-    }
-
-    if ((timer->long_target > _long_cnt) || !_this_high_period(target)) {
-        DEBUG("xtimer_set_absolute(): the timer doesn't fit into the low-level timer's mask.\n");
-        _add_timer_to_long_list(&long_list_head, timer);
-    }
-    else {
-        if (_xtimer_lltimer_mask(now) >= target) {
-            DEBUG("xtimer_set_absolute(): the timer will expire in the next timer period\n");
-            _add_timer_to_list(&overflow_list_head, timer);
-        }
-        else {
-            DEBUG("timer_set_absolute(): timer will expire in this timer period.\n");
-            _add_timer_to_list(&timer_list_head, timer);
-
-            if (timer_list_head == timer) {
-                DEBUG("timer_set_absolute(): timer is new list head. updating lltimer.\n");
-                _lltimer_set(target);
-            }
-        }
-    }
-
-    irq_restore(state);
-
-    return res;
+    _lltimer_ongoing = true;
 }
 
 static void _add_timer_to_list(xtimer_t **list_head, xtimer_t *timer)
 {
-    while (*list_head && (*list_head)->target <= timer->target) {
+    while (*list_head
+        && (*list_head)->start_time + (*list_head)->offset - timer->start_time <= timer->offset) {
```

Comment:
Could you change it to:
` && (*list_head)->start_time + (*list_head)->offset <=  timer->start_time + timer->offset) {`

Reply:
same as above.

Reply:
same as above